### PR TITLE
feat: Add support for xyn files

### PR DESCRIPTION
**Pull request by tim-vd-aardweg**
Fixes: #459

### CODE REVIEW

**Comment by tim-vd-aardweg** [comment]
There are 2 code smells left that I believe we can ignore. I do not like to put those strings in a constant.

File: hydrolib/core/dflowfm/xyn/name_extrator.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
+class NameExtractor:
```
Comment (arthurvd) [inline hydrolib/core/dflowfm/xyn/name_extrator.py:1]: Maybe this whole class could be replaced by the following regexp?
```python
prog=re.compile("^\s*('?)([^'#]+)\\1\s*#?.*$")
if (match := prog.match(text) is not None):
    return match.group(2)
# ...
```
See here some examples on which it works:
```
In [30]: print(prog.match("'somevalue in quotes' #comment").group(2))
somevalue in quotes

In [31]: print(prog.match("  somevalue_without_quotes #comment").group(2))
somevalue_without_quotes

In [32]: print(prog.match("  somevalue_without_quotes_and_no_comment  ").group(2))
somevalue_without_quotes_and_no_comment

In [33]: print(prog.match("'somevalue in quotes, but no comment'  ").group(2))
somevalue in quotes, but no comment

In [34]: print(prog.match("'somevalue in quotes, but no comment and missing closing quote  "))
None
```

Comment (arthurvd) [reply hydrolib/core/dflowfm/xyn/name_extrator.py:1]: Upon second checking: no comments at the end of a data line are supported at all in .xyn files, so please remove the `#` handling (either in your own code, or in my above suggestion).

File: hydrolib/core/dflowfm/xyn/serializer.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+from typing import Dict
+
+from hydrolib.core.basemodel import SerializerConfig
+
+
+class XYNSerializer:
+    @staticmethod
+    def serialize(path: Path, data: Dict, config: SerializerConfig) -> None:
+        """
+        Serializes the XYN data to the file at the specified path.
```
Comment (arthurvd) [inline hydrolib/core/dflowfm/xyn/serializer.py:11]:
```suggestion
        Serializes the observation point data to an .xyn file at the specified path.
```

File: tests/data/input/obsfile_cases/single_ini/obs.ini
Comment (arthurvd) [inline tests/data/input/obsfile_cases/single_ini/obs.ini:1]: this .ini duplicates with the case with both xyn and ini. Maybe only have the mdu's different, and refer to the same .ini?

File: tests/data/input/obsfile_cases/single_xyn/obs.xyn
Comment (arthurvd) [inline tests/data/input/obsfile_cases/single_xyn/obs.xyn:1]: this duplicates with the xyn from the case with both ini and xyn. Maybe only include the MDU and refer to the existing xyn file?

**Review by arthurvd — CHANGES_REQUESTED** [review]
Hi @tim-vd-aardweg , please see my review comments.

File: tests/data/input/obsfile_cases/single_ini/obs.ini
Comment (tim-vd-aardweg) [reply tests/data/input/obsfile_cases/single_ini/obs.ini:1]: I am a bit conflicted here. I generally do not want to use the  same test data (file) for different tests (with different purposes). Let's discuss it today.

Comment (tim-vd-aardweg) [reply tests/data/input/obsfile_cases/single_ini/obs.ini:1]: Discussed this with Arthur and we will leave it as is for now.

File: tests/data/input/obsfile_cases/single_xyn/obs.xyn
Comment (tim-vd-aardweg) [reply tests/data/input/obsfile_cases/single_xyn/obs.xyn:1]: Discussed this with Arthur and we will leave it as is for now.

File: hydrolib/core/dflowfm/xyn/name_extrator.py (hunk 1 above)
Comment (tim-vd-aardweg) [reply hydrolib/core/dflowfm/xyn/name_extrator.py:4]: I have spent some time trying to get it to work with a single regex, but I could not get it to work :(

File: hydrolib/core/dflowfm/xyn/models.py [hunk 3]
```diff
@@ -46,7 +46,7 @@ def _ext(cls) -> str:
 
     @classmethod
     def _filename(cls) -> str:
-        return "sample"
+        return "stations_obs.xyn"
```
Comment (arthurvd) [inline hydrolib/core/dflowfm/xyn/models.py:49]:
```suggestion
        return "stations_obs"

    @abstractclassmethod
    def _ext(cls) -> str:
        return ".xyn"
```

**Review by arthurvd — CHANGES_REQUESTED** [review]
All fine now, except for one small new thing that I overlooked last time. Can you consider/commit my suggestion? After that I will merge.

**Review by arthurvd — APPROVED** [review]

**Pull request merged**